commit 80a42122a023df2e716738fd06bef2d2a49dce8f
Author: Sage Weil <sweil@redhat.com>
Date:   Thu Jul 20 16:35:20 2017 -0400

    mon: 'versions' command to show running versions for daemons of all types
    
    Easier!
    
    Signed-off-by: Sage Weil <sage@redhat.com>

diff --git a/src/mgr/ServiceMap.h b/src/mgr/ServiceMap.h
index 5137f9f84f2..cd12133b871 100644
--- a/src/mgr/ServiceMap.h
+++ b/src/mgr/ServiceMap.h
@@ -51,6 +51,19 @@ struct ServiceMap {
 	 << " active";
       return ss.str();
     }
+
+    void count_metadata(const string& field,
+			std::map<std::string,int> *out) const {
+      for (auto& p : daemons) {
+	auto q = p.second.metadata.find(field);
+	if (q == p.second.metadata.end()) {
+	  (*out)["unknown"]++;
+	} else {
+	  (*out)[q->second]++;
+	}
+      }
+    }
+
   };
 
   epoch_t epoch = 0;
